### PR TITLE
Fix #476: Slow readback of source channels with empty ROI

### LINKED ISSUE #476: Slow readback of source channels with empty ROI
Reported by @steiner- and @egparedes

### PR DESCRIPTION
**Pull request by eile**

### CODE REVIEW

File: doc/Changelog.md [hunk 1]
```diff
@@ -3,6 +3,8 @@ Changelog {#Changelog}
 
 # git master {#master}
 
+* [476](https://github.com/Eyescale/Equalizer/issues/476):
+  Slow readback of source channels with empty ROI
```
Comment (tribal-tec) [inline doc/Changelog.md]: Fix slow readback...

File: eq/channel.cpp [hunk 2]
```diff
@@ -804,8 +799,11 @@ void Channel::declareRegion( const PixelViewport& region )
 
 PixelViewport Channel::getRegion() const
 {
-    PixelViewport region;
-    for( size_t i = 0; i < _impl->regions.size(); ++i )
+    if( _impl->regions.empty( ))
+        return PixelViewport();
```
Comment (tribal-tec) [inline eq/channel.cpp]: it still the same as before?!?

Comment (eile) [inline eq/channel.cpp]: Come to think of it, the fix is different. Merge( invalid, empty pvp ) should produce an empty pvp.

Comment (tribal-tec) [inline eq/channel.cpp]: I see. Was just focusing on the described empty ROI, which to me translated to regions.empty().

Comment (eile) [inline eq/channel.cpp]: The real bug is that merge(invalid, empty) should yield empty and not invalid. Fix will come soon.

**Comment by eile** [comment]
updated

**Pull request merged**